jarbas/frontend/elm/Reimbursement/Fields.elm

```diff
@@ -1,170 +1,319 @@
-isSearchable : ( String, String ) -> Bool
-isSearchable ( field, label ) =
-    if field == "page" then
-        True
-    else
-        List.member field
-            [ "applicantId"
-            , "cnpjCpf"
-            , "documentId"
-            , "issueDateEnd"
-            , "issueDateStart"
-            , "month"
-            , "subquotaId"
-            , "year"
-            ]
 
+labelToUrl : Label -> String
+labelToUrl label =
+    List.Extra.find (Tuple.first >> (==) label) searchableLabels
+        |> Maybe.map (Tuple.second)
+        |> Maybe.withDefault ""
 
-sets : Language -> List ( Int, ( String, List String ) )
-sets lang =
-    List.indexedMap
-        (,)
-        [ ( translate lang SearchFieldsetCongressperson
-          , [ "applicantId"
-            , "year"
-            , "month"
-            , "subquotaId"
-            , "cnpjCpf"
-            , "issueDateStart"
-            , "issueDateEnd"
+
+urlToLabel : String -> Label
+urlToLabel url =
+    List.Extra.find (Tuple.second >> (==) url) searchableLabels
+        |> Maybe.map (Tuple.first)
+        |> Maybe.withDefault Empty
+
+
+sets : List ( Int, ( TranslationId, List Label ) )
+sets =
+    List.indexedMap (,)
+        [ ( SearchFieldsetCongressperson
+          , [ ApplicantId
+            , Year
+            , Month
+            , SubquotaId
+            , CnpjCpf
+            , IssueDateStart
+            , IssueDateEnd
             ]
           )
-        , ( translate lang SearchFieldsetReimbursement
-          , [ "applicantId", "year", "documentId" ]
+        , ( SearchFieldsetReimbursement
+          , [ ApplicantId, Year, DocumentId ]
           )
         ]
 
 
 {-| Filter to get numbers only fields:
 
-    >>> isNumeric "year"
+    >>> isNumeric Year
     True
 
-    >>> isNumeric "format"
+    >>> isNumeric Company
     False
 
 -}
-isNumeric : String -> Bool
-isNumeric field =
-    List.member field
-        [ "applicantId"
-        , "cnpjCpf"
-        , "congresspersonId"
-        , "documentId"
-        , "month"
-        , "subquotaId"
-        , "year"
-        ]
+isNumeric : Label -> Bool
+isNumeric label =
+    case label of
+        ApplicantId ->
+            True
 
+        CnpjCpf ->
+            True
 
-isDate : String -> Bool
-isDate field =
-    List.member field
-        [ "issueDateStart"
-        , "issueDateEnd"
-        ]
+        CongresspersonId ->
+            True
+
+        DocumentId ->
+            True
+
+        Month ->
+            True
+
+        SubquotaId ->
+            True
+
+        Year ->
+            True
+
+        _ ->
+            False
+
+
+isDate : Label -> Bool
+isDate label =
+    case label of
+        IssueDateStart ->
+            True
+
+        IssueDateEnd ->
+            True
+
+        _ ->
+            False
+
+
+getValue : Field -> String
+getValue (Field _ value) =
+    value
+
+
+getLabel : Field -> Label
+getLabel (Field label _) =
+    label
+
+
+getLabelTranslation : Language -> Field -> String
```

Comment:
Do we really need to move translations out of the `Internationalization` module? I know `Internationalization.elm` is a long file that  we could break down, but somehow having all translations in the same module is a good thing (even if we break this module down in submodules).